app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -449,12 +477,46 @@ public void onActivityResult(int requestCode, int resultCode, Intent intent)
             format = result.getFormatName();
         }
 
-        if(requestCode == SELECT_BARCODE_REQUEST && resultCode == Activity.RESULT_OK)
+        if (resultCode == Activity.RESULT_OK)
         {
-            Log.i(TAG, "Received barcode information from typing it");
+            if(requestCode == SELECT_BARCODE_REQUEST)
+            {
+                Log.i(TAG, "Received barcode information from typing it");
 
-            contents = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_CONTENTS);
-            format = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_FORMAT);
+                contents = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_CONTENTS);
+                format = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_FORMAT);
+            }
+            else if (requestCode == SELECT_IMAGE_REQUEST)
+            {
+                Log.i(TAG, "Received barcode image");
+
+                Bitmap bitmap = null;
+                try {
+                    bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), intent.getData());
+                } catch (IOException e) {
+                    Log.e(TAG, "Error getting the image data");
+                    e.printStackTrace();
+                    return;
+                }
+
+                // In order to decode it, the Bitmap must first be converted into a pixel array...
+                int[] intArray = new int[bitmap.getWidth()*bitmap.getHeight()];
```

Comment:
Do you know what the failure mode is if a user selects an image which "is too big"? For example, in generating the barcode for display I've sometimes encountered issues with memory:

https://github.com/brarcher/loyalty-card-locker/blob/master/app/src/main/java/protect/card_locker/BarcodeImageWriterTask.java#L152

Reply:
I don't actually know... I looked around and I'm still not sure about an OOM exception, however I did stumble upon people having issues with codes in huge images not being recognized.

I'll do some more testing and maybe add a downscale step before processing.